dnf/cli/output.py

```diff
@@ -2250,3 +2261,119 @@ def progressbar(current, total, name=None):
         sys.stdout.write('\n')
 
     sys.stdout.flush()
+
+class PackageFormatter(object):
+    """
+    # :api
+    Wrapper for dnf.package.Package, so we can control formatting
+    in --queryformat
+    """
+    def __init__(self, pkg):
+        self._pkg = pkg
+
+    def __format__(self, fmt):
+        return self.format(fmt)
+
+    def _attr(self, name):
+        name = name.lower()
+        if name not in QUERY_TAGS:
+            raise AttributeError("Unknown tag '%s'" % name)
+        return getattr(self, name, "(none)")
+
+    def __getattr__(self, attr):
+        return getattr(self._pkg, attr)
+
+    def _replace_tag(self, match):
+        subfmt = "%%%ss" % match.group(1)
+        name, fmt_fn = self._tag_format(match.group(2))
+        value = fmt_fn(self._attr(name))
+        return subfmt % (value,)
+
+    def _replace_ternary(self, match):
+        if self._attr(match.group(1)):
+            tag = match.group(2)
+        else:
+            tag = match.group(4) if match.groups() > 2 else ""
+        return tag
+
+    def _tag_format(self, tag):
+        sep = tag.find(':')
+        if sep > -1:
+            name = tag[:sep]
+            tag_format = tag[sep+1:]
+            if tag_format not in QUERY_TAG_FORMATS:
+                raise AttributeError("Unknown tag format '%s'" % tag_format)
```

Comment:
I think it should rather raise `ValueError` like `str.format` does. AttributeError implies that there is no attribute called `_tag_format`.